src/createAnimatedComponent.tsx

```diff
@@ -66,21 +86,85 @@ function flattenArray(array) {
   return resultArr;
 }
 
-export default function createAnimatedComponent(Component, options = {}) {
+interface AnimatedProps extends Record<string, unknown> {
+  viewDescriptors?: ViewDescriptorsSet;
+  viewsRef?: ViewRefSet<unknown>;
+  initial?: SharedValue<StyleProps>;
+}
+
+export type AnimatedComponentProps<P extends Record<string, unknown>> = {
+  [K in keyof P]: P[K];
```

Comment:
I don't get this type.

Reply:
It simply merges props of the given component with props that are used by our AnimatedComponent.